valhalla/filesystem.h

```diff
@@ -416,7 +419,11 @@ inline void resize_file(const path& p, std::uintmax_t new_size) {
 }
 
 inline bool remove(const path& p) {
-  return ::remove(p.c_str()) == 0;
+  bool success = ::remove(p.c_str()) == 0;
+  if (!success) {
+    success = errno == ENOENT;
+  }
+  return success;
```

Comment:
i think this breaks compatibility with stls implementation. i'd prefer that not to be the case. eventually we're going to move to c++17 and this will be a rude gotcha, what do you think @ktatterso ? is this fixing a bug that we are also running into? also it looks like technically we should implement something like this to meet the spec:

```suggestion
  // if it failed for any reason other than the path not existing then throw
  if (remove(p.c_str()) != 0 && errno != ENOENT) {
    throw std::runtime_error("filesystem_error filesytem::remove");
  }
  // otherwise no error means success or not existing means failure to remove
  return errno == 0;
```

Reply:
I added this check because of a couple reasons.  First, it mirrors the check you added for mkdir, which basically says "if the folder already exists return success".  Hence, the remove logic says, "if you want to remove a directory that doesn't exist return success."

The other reason was I wrote the unit test for concurrent subdirectory remove to expect all threads to succeed.  That expectation could be adapted to whatever we desire, though.

Honestly, I like approach of returning success if asked to remove something that doesn't exist. But writing C++17 compatible code is more important than my feelings :).  As a result, I think these changes to remove() should be reverted - the original implementation was "correct" from the C++17 perspective.

@kevinkreiser I can make these changes if you want?  And update the unit test accordingly.

Reply:
the orginal was correct except it didnt throw when it had an error other than enoent. it was missing the `throw`. :+1: on making the changes if you're good with it!

Reply:
Sounds good.